api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/ContrastAccountServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+        BasicSessionCredentials credentials = credentialProvider.getBaseAccCredentials();
+        String region = System.getenv("REGION");
+        AWSSecretsManager secretClient = AWSSecretsManagerClientBuilder
+                .standard()
+                .withCredentials(new AWSStaticCredentialsProvider(credentials))
+                .withRegion(region).build();
+        String secretId = getSecretId(accountData.getContrastOrganizationId(), CONTRAST);
+        ListSecretsRequest listSecretsRequest = new ListSecretsRequest()
+                .withFilters(new Filter().withKey("name").withValues(secretId))
+                .withIncludePlannedDeletion(true);
+        ListSecretsResult getListResponse = secretClient.listSecrets(listSecretsRequest);
+        if (getListResponse.getSecretList().size() > 0) {
+            LOGGER.info("Secret already exists. Deleting in progress.");
+            validationResponse = new AccountValidationResponse();
+            validationResponse.setValidationStatus(FAILURE);
+            validationResponse.setMessage("Prior account is being deleted. Try again in a couple of minutes.");
+            return validationResponse;
+        }
+        CreateSecretRequest createRequest = new CreateSecretRequest()
+                .withName(secretId).withSecretString(getContrastSecretString(accountData));
+        CreateSecretResult createResponse = secretClient.createSecret(createRequest);
+        LOGGER.info("Create secret response: {}", createResponse);
+        String accountId = UUID.randomUUID().toString();
+        createAccountInDb(accountData.getContrastOrganizationId(), organizationName, CONTRAST, accountData.getCreatedBy());
+        updateConfigProperty(CONTRAST_ENABLED, TRUE, JOB_SCHEDULER);
+        sendSQSMessage(CONTRAST, accountData.getContrastOrganizationId());
+        validationResponse = new AccountValidationResponse();
+        validationResponse.setValidationStatus(SUCCESS);
+        validationResponse.setAccountId(accountId);
+        validationResponse.setAccountName(organizationName);
+        validationResponse.setType(CONTRAST);
+        validationResponse.setMessage("Account added successfully. ID: " + accountId);
+        return validationResponse;
+    }
+
+    private AccountValidationResponse validateRequestData(CreateAccountRequest accountData) {
+        AccountValidationResponse response = new AccountValidationResponse();
+        List<String> missingParams = new ArrayList<>();
+        if (StringUtils.isEmpty(accountData.getContrastOrganizationId())) {
+            missingParams.add("organizationId");
+        }
+        if (StringUtils.isEmpty(accountData.getContrastServiceKey())) {
+            missingParams.add("serviceKey");
+        }
+        if (StringUtils.isEmpty(accountData.getContrastApiKey())) {
+            missingParams.add("apiKey");
+        }
+        if (StringUtils.isEmpty(accountData.getContrastUserId())) {
+            missingParams.add("userId");
```

Comment:
Make names human-friendly to read - this message will be preset in UI

Reply:
done